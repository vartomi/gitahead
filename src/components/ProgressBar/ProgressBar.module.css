.container {
    text-align: center;
}

.container.compact {
    position: absolute;
    top: 12px;
    right: 8px;
}

.animation {
    -webkit-animation:spin 3s linear infinite;
    -moz-animation:spin 3s linear infinite;
    animation:spin 3s linear infinite;
}

@-moz-keyframes spin { 100% { -moz-transform: rotate(360deg); } }
@-webkit-keyframes spin { 100% { -webkit-transform: rotate(360deg); } }
@keyframes spin { 100% { -webkit-transform: rotate(360deg); transform:rotate(360deg); } }