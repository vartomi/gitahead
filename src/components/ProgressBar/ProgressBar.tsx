import React from 'react';
import styles from './ProgressBar.module.css';
import { ReactComponent as Logo } from '../../assets/logo.svg';
import classNames from 'classnames';

type ProgressBarProps = {
    compact?: boolean;
}

const ProgressBar: React.FC<ProgressBarProps> = ({ compact }) => {
    const size: number = compact ? 24 : 64;
    return (
        <div className={classNames(styles.container, { [styles.compact]: compact })}>
            <Logo height={size} width={size} className={styles.animation} />
            {!compact && <div>Loading...</div>}
        </div>
    );
}

export default ProgressBar;
