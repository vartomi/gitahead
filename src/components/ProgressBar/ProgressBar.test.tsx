import { shallow } from 'enzyme'
import ProgressBar from './ProgressBar';
import React from 'react';
import { ReactComponent as Logo } from '../../assets/logo.svg';

describe('ProgressBar component', () => {
    it('should be rendered with Logo', () => {
        const wrapper = shallow(<ProgressBar />)
        expect(wrapper.find(Logo).length).toBeTruthy();
    });

    it('should render snapshot', () => {
        const wrapper = shallow(<ProgressBar />)
        expect(wrapper).toMatchSnapshot();
    });

    it('should render compact snapshot', () => {
        const wrapper = shallow(<ProgressBar compact={true} />)
        expect(wrapper).toMatchSnapshot();
    });
});