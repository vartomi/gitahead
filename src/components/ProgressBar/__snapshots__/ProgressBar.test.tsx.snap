// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar component should render compact snapshot 1`] = `
<div
  className="container compact"
>
  <ForwardRef(SvgLogo)
    className="animation"
    height={24}
    width={24}
  />
</div>
`;

exports[`ProgressBar component should render snapshot 1`] = `
<div
  className="container"
>
  <ForwardRef(SvgLogo)
    className="animation"
    height={64}
    width={64}
  />
  <div>
    Loading...
  </div>
</div>
`;
